### PR TITLE
Pull Request for Issue #3186 Content Processor Updates

### PR DESCRIPTION
**Pull request by Alturos**
Updated ContentTypeReaderManager to make type resolutions for ContentTypeReaders significantly more robust.

### CODE REVIEW

**Comment by tomspilman** [comment]
@Alturos - Can you merge the latest changes from develop into your branch?

File: MonoGame.Framework/Content/ContentTypeReaderManager.cs [hunk 1]
```diff
@@ -35,40 +35,57 @@ namespace Microsoft.Xna.Framework.Content
...
+        /// 1 - Name up to `
+        /// 2 - Number of Generic Parameters
+        /// 3 - [
+        /// 4 - Collection of Generic types
+        /// 5 - ]
+        /// 6 - Optional ', AssemblyName'
+        /// 
+        /// Windows 8 RT doesn't appear to support Compiled Regex.
+        /// </summary>
+#if (WINDOWS_STOREAPP || WINDOWS_PHONE)
```
Comment (KonajuGames) [inline MonoGame.Framework/Content/ContentTypeReaderManager.cs]: Would this also need to include Windows UWP?

Comment (Alturos) [inline MonoGame.Framework/Content/ContentTypeReaderManager.cs]: I initially believed so, but after recalling that the Win10 build passed, I took another look. UAP appears to have the Compiled support that WIn8 did not.

**Comment by Alturos** [comment]
@tomspilman Done!

**Comment by tomspilman** [comment]
We have conflicts here and we expect some big changes to the pipeline soon.  So closing this.

**Pull request closed without merging**